### DISCUSSION TITLE
How To Disconnect

### DISCUSSION THREAD

**Discussion by mob-aaron** [miguelgrinberg/flask-sock#98] [Q&A]
I would like to disconnect from clients that don't follow the established protocol for my application.

```python
from flask_sock import Sock
sock = Sock()
@sock.route('/api/progress', methods=['GET'])
def get_progress(ws):
    data = ws.receive()
    if not check_data(data):
        ws.send("DISCONNECTING")
        ws.disconnect()
```

fails with `AttributeError: 'Server' object has no attribute 'disconnect'`. How do I disconnect a client?

**Comment by miguelgrinberg** [comment]
To disconnect you can call `ws.close()` or just return from the handler function.
